style-guide/principles.md

```diff
@@ -0,0 +1,51 @@
+# Guiding principles and rationale
+
+When deciding on style guidelines, the style team tried to be guided by the
+following principles (in rough priority order):
+
+* readability
+    - scan-ability
+    - avoiding misleading formatting
+    - accessibility - readable and editable by users using the the widest
+      variety of hardware, including non-visual accessibility interfaces
+    - readability of code when quoted in rustc error messages
+
+* aesthetics
+    - sense of 'beauty'
+    - consistent with other languages/tools
+
+* specifics
+    - compatibility with version control practices - preserving diffs,
+      merge-friendliness, etc.
+    - preventing right-ward drift
+    - minimising vertical space
+
+* application
+    - ease of manual application
+    - ease of implementation (in Rustfmt, and in other tools/editors/code generators)
+    - internal consistency
+    - simplicity of formatting rules
+
+
+## Overarching guidelines
+
+Prefer block indent over visual indent. E.g.,
```

Comment:
One of the main goals of these formatting guidelines is to increase uniformity across Rust code. That helps people read code consistently, and contribute to a variety of projects.

If you're seeking a rustfmt option, the place to discuss that would be in rustfmt, not in the formatting guidelines, which state the standard style, and mostly don't discuss non-standard styles. (They do mention things like indentation width, just because that tends to be such a hot question, but even there the guidelines define a standard style.)

That said, visual vs block is something I very much hope rustfmt doesn't offer an option for.